backend/config/corpora-api.yml

```diff
@@ -286,6 +281,59 @@ paths:
         "413":
           $ref: "#/components/responses/413"
 
+    put:
+      summary: Reupload a new version of an existing dataset
+      tags:
+        - collections
+        - datasets
+      description: >-
```

Comment:
Should we differentiate between how this acts for a published v private collection? It'll replace the dataset in the private version of the collection if they are doing a revision but I'm not sure thats clear from 
```Once the upload of the replacement dataset is complete, the previous version of the dataset is replaced and is
        no longer available for downloading through this collection```

Reply:
How about 
>        The existing dataset must be in an error start or completely uploaded before a replacement can be attempted. Once the upload of the replacement dataset is complete, the previous version of the dataset is removed and is no longer available for downloading through this collection. The replacement datasets are not publicly available until the collection revision is published. The old datasets are available in the public collection until the collection revision is published.